commander3/src/comm_tod_WMAP_mod.f90

```diff
@@ -19,469 +19,1237 @@
+            end do
+         end do
+      end do
+      if (self%myid == 0) map_full = map_full/self%ndet
+      deallocate (m_buf)
+!      call mpi_finalize(ierr)
+!      stop
+
+      allocate(procmask(0:npix-1))
+      procmask = 0
+      do i = 1, size(self%procmask%map(:,1))
+         procmask(self%procmask%info%pix(i)) = nint(self%procmask%map(i-1,1))
+      end do
+      call mpi_allreduce(mpi_in_place, procmask, size(procmask), MPI_INTEGER, MPI_SUM, self%info%comm, ierr)
+      !where (procmask .ge. 1)
+      !    procmask = 1
+      !end where
+
+      call wall_time(t2); t_tot(9) = t2 - t1
+
+      ! Compute far sidelobe Conviqt structures
       call wall_time(t1)
-      do j = 1, ndet
-         if (.not. self%scans(i)%d(j)%accept) cycle
-         call self%decompress_pointing_and_flags(i, j, pix(:,j), &
-              & psi(:,j), flag(:,j))
+      do i = 1, self%ndet
+         if (.not. correct_sl) exit
+         !TODO: figure out why this is rotated
+         call map_in(i,1)%p%YtW()  ! Compute sky a_lms
+         self%slconv(i)%p => comm_conviqt(self%myid, self%comm_shared, &
+              & self%myid_inter, self%comm_inter, self%slbeam(i)%p%info%nside, &
+              & 100, 3, 100, self%slbeam(i)%p, map_in(i,1)%p, 2)
       end do
-      call self%symmetrize_flags(flag)
-      call wall_time(t2); t_tot(11) = t_tot(11) + t2-t1
-      !call update_status(status, "tod_decomp")
-
-      ! Construct sky signal template
-      !TODO: we will want this but I don't know if it will be the same code or
-      !not
-      !call wall_time(t1)
-      !if (do_oper(bin_map) .or. do_oper(prep_relbp)) then
-      !   call project_sky(self, map_sky(:,:,:,1), pix, psi, flag, &
-      !        & sprocmask%a, i, s_sky, mask, s_bp=s_bp)
-      !else
-      !   call project_sky(self, map_sky(:,:,:,1), pix, psi, flag, &
-      !        & sprocmask%a, i, s_sky, mask)
-      !end if
-      !if (do_oper(prep_relbp)) then
-      !   do j = 2, ndelta
-      !      call project_sky(self, map_sky(:,:,:,j), pix, psi, flag, &
-      !           & sprocmask2%a, i, s_sky_prop(:,:,j), mask2, s_bp=s_bp_prop(:,:,j))
-      !   end do
-      !else if (do_oper(prep_absbp)) then
-      !   do j = 2, ndelta
-      !      call project_sky(self, map_sky(:,:,:,j), pix, psi, flag, &
-      !           & sprocmask2%a, i, s_sky_prop(:,:,j), mask2)
-      !   end do
-      !end if
-      if (main_iter == 1 .and. self%first_call) then
-         do j = 1, ndet
-            if (all(mask(:,j) == 0)) self%scans(i)%d(j)%accept = .false.
-            if (self%scans(i)%d(j)%sigma0 <= 0.d0) self%scans(i)%d(j)%accept = .false.
+      call wall_time(t2); t_tot(13) = t2-t1
+
+      call update_status(status, "tod_init")
+      call wall_time(t3)
+      do_oper             = .true.
+      allocate (M_diag(0:npix-1, nmaps+1))
+      allocate ( b_map(0:npix-1, nmaps, nout))
+      M_diag = 0d0
+      b_map = 0d0
+      ! There are five main iterations, for imbalance, absolute calibration, 
```

Comment:
Again, I think redability would improve dramatically if we did something other than this system